der/src/decoder.rs

```diff
@@ -283,6 +283,13 @@ impl<'a> Decoder<'a> {
         Ok(self.bytes.ok_or(ErrorKind::Failed)?.len())
     }
 
+    /// Obtain a slice of bytes contain a complete TLV production suitable for parsing later.
+    pub fn tlv_slice(&mut self) -> Result<&'a [u8]> {
```

Comment:
Generally other methods in these crates that operate over a slice of bytes are named `*_bytes`. How about:

```suggestion
    pub fn tlv_bytes(&mut self) -> Result<&'a [u8]> {
```